core-rust/state-manager/src/staging/mod.rs

```diff
@@ -63,6 +63,41 @@
  */
 
 mod cache;
+mod result;
 mod stage_tree;
 
+use crate::accumulator_tree::storage::ReadableAccuTreeStore;
+use crate::{ReceiptHash, TransactionHash};
+use radix_engine::ledger::ReadableSubstateStore;
+use radix_engine_interface::api::types::SubstateOffset;
+use radix_engine_stores::hash_tree::tree_store::{ReNodeModulePayload, ReadableTreeStore};
+
 pub use cache::*;
+pub use result::*;
+
+pub trait ReadableLayeredTreeStore:
+    ReadableTreeStore<ReNodeModulePayload> + ReadableTreeStore<SubstateOffset>
+{
+}
+impl<T: ReadableTreeStore<ReNodeModulePayload> + ReadableTreeStore<SubstateOffset>>
+    ReadableLayeredTreeStore for T
+{
+}
+
+pub trait ReadableTxAndRxTreeStore:
+    ReadableAccuTreeStore<u64, TransactionHash> + ReadableAccuTreeStore<u64, ReceiptHash>
```

Comment:
whoa, whoa, this one actually has a rust doc :D

In its original PR there is an explanation where the "accumulator tree" name came from: https://github.com/radixdlt/babylon-node/pull/347#discussion_r1128008100

I still think it is a pretty good name to adopt from there. Especially since we've been discussing replacing our accumulator hash with that tree.

I guess you want to expand the name into `ReadableAccumulatorTreeStore`...?